include/mumble/Message.hpp

```diff
@@ -67,6 +67,72 @@ struct Message {
 	Timestamp timestamp;
 };
 
+namespace udp {
+	struct Message : public mumble::Message {
+		struct Audio;
+		struct Ping;
+
+		enum class Type : uint8_t { Audio, Ping, Unknown = std::numeric_limits< uint8_t >::max() };
+
+		Message()          = default;
+		virtual ~Message() = default;
+
+		Protocol protocol() const { return Protocol::UDP; }
+
+		virtual Type type() const { return Type::Unknown; }
```

Comment:
Wouldn't it make more sense to make the base `Message` class abstract by turning this into a pure virtual function? Then we also don't have to define the `UNKNOWN` type, which will become annoying, when handling the different types via a switch statement.